File: Sources/Api/SmartLoginAPI.m
//
//  SmartLoginAPI.m
//  LoginRadiusSDK
//
//  Created by LoginRadius on 30/07/18.
//

#import "SmartLoginAPI.h"

@implementation SmartLoginAPI
+ (instancetype)smartLoginInstance{
    static dispatch_once_t onceToken;
    static SmartLoginAPI *instance;
    
    dispatch_once(&onceToken, ^{
        instance = [[SmartLoginAPI alloc] init];
    });
    
    return instance;
}

- (void)smartLoginWithEmail:(NSString *)email clientguid:(NSString *)clientguid smartloginemailtemplate:(NSString *)smartloginemailtemplate welcomeemailtemplate:(NSString *)welcomeemailtemplate redirecturl:(NSString *)redirecturl completionHandler:(LRAPIResponseHandler)completion {
    NSString *smartlogin_emailtemplate = smartloginemailtemplate ? smartloginemailtemplate: @"";
    NSString *welcome_emailtemplate = welcomeemailtemplate ? welcomeemailtemplate: @"";
    NSString *redirect_url = redirecturl ? redirecturl: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/smartlogin"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"email": email,
                                             @"clientguid": clientguid,
                                             @"smartloginemailtemplate":smartlogin_emailtemplate,
                                             @"welcomeemailtemplate":welcome_emailtemplate,
                                             @"redirecturl":redirect_url
                                             }
                         completionHandler:completion];
    
}

-(void)smartLoginWithUsername:(NSString *)username clientguid:(NSString *)clientguid smartloginemailtemplate:(NSString *)smartloginemailtemplate welcomeemailtemplate:(NSString *)welcomeemailtemplate redirecturl:(NSString *)redirecturl completionHandler:(LRAPIResponseHandler)completion{
    NSString *smartlogin_emailtemplate = smartloginemailtemplate ? smartloginemailtemplate: @"";
    NSString *welcome_emailtemplate = welcomeemailtemplate ? welcomeemailtemplate: @"";
    NSString *redirect_url = redirecturl ? redirecturl: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/smartlogin"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"username": username,
                                             @"clientguid": clientguid,
                                             @"smartloginemailtemplate":smartlogin_emailtemplate,
                                             @"welcomeemailtemplate":welcome_emailtemplate,
                                             @"redirecturl":redirect_url
                                             }
                         completionHandler:completion];
    
}

-(void)smartLoginPingWithClientguid:(NSString *)clientguid completionHandler:(LRAPIResponseHandler)completion{
    
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/smartlogin/ping"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"clientguid": clientguid
                                             }
                         completionHandler:completion];
    
}

-(void)smartAutoLoginWithVerificationToken:(NSString *)verificationtoken welcomeemailtemplate:(NSString *)welcomeemailtemplate completionHandler:(LRAPIResponseHandler)completion{
    NSString *welcome_emailtemplate = welcomeemailtemplate ? welcomeemailtemplate: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/email/smartlogin"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"verificationtoken": verificationtoken,
                                             @"welcomeemailtemplate": welcome_emailtemplate
                                             }
                         completionHandler:completion];
    
}

@end

File: Sources/SmartLoginAPI.swift
//
//  SmartLoginAPI.swift
//  LoginRadiusSwift SDK
//
//  Created by Megha Agarwal.
//  Copyright © 2023 LoginRadius Inc. All rights reserved.
//

import Foundation



public class SmartLoginAPI {
    public static let shared = SmartLoginAPI()
    
    public init() {}
    
    public func smartLoginWithEmail(email: String, clientguid: String, smartloginemailtemplate: String?, welcomeemailtemplate: String?, redirecturl: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let smartlogin_emailtemplate = smartloginemailtemplate ?? ""
        let welcome_emailtemplate = welcomeemailtemplate ?? ""
        let redirect_url = redirecturl ?? ""
        
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/smartlogin", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "email": email,
            "clientguid": clientguid,
            "smartloginemailtemplate": smartlogin_emailtemplate,
            "welcomeemailtemplate": welcome_emailtemplate,
            "redirecturl": redirect_url
        ], completionHandler: completionHandler)
    }
    
    public func smartLoginWithUsername(username: String, clientguid: String, smartloginemailtemplate: String?, welcomeemailtemplate: String?, redirecturl: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let smartlogin_emailtemplate = smartloginemailtemplate ?? ""
        let welcome_emailtemplate = welcomeemailtemplate ?? ""
        let redirect_url = redirecturl ?? ""
        
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/smartlogin", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "username": username,
            "clientguid": clientguid,
            "smartloginemailtemplate": smartlogin_emailtemplate,
            "welcomeemailtemplate": welcome_emailtemplate,
            "redirecturl": redirect_url
        ], completionHandler: completionHandler)
    }
    
    public func smartLoginPingWithClientguid(clientguid: String, completionHandler: @escaping LRAPIResponseHandler) {
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/smartlogin/ping", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "clientguid": clientguid
        ], completionHandler: completionHandler)
    }
    
    public func smartAutoLoginWithVerificationToken(verificationtoken: String, welcomeemailtemplate: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let welcome_emailtemplate = welcomeemailtemplate ?? ""
        
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/email/smartlogin", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "verificationtoken": verificationtoken,
            "welcomeemailtemplate": welcome_emailtemplate
        ], completionHandler: completionHandler)
    }
}
